noir-projects/aztec-nr/aztec/src/utils/bytes.nr

```diff
@@ -0,0 +1,120 @@
+        // in `bytes_to_fields` so we can safely convert it to 31 bytes.
+        let src: [u8; 31] = input[field_index].to_be_bytes();
+
+        // Since some of the bytes might not be occupied (if the source value requiring less than 31 bytes),
+        // we have to compute the start index from which to copy.
+        let remaining_bytes = N - field_index * 31;
+        let src_start_index = if remaining_bytes < 31 {
+            // If the remaining bytes are less than 31, we only copy the remaining bytes
+            31 - remaining_bytes
+        } else {
+            0
+        };
+
+        for i in 0..31 {
+            let byte_index = field_index * 31 + i;
+            if byte_index < N {
+                dst[byte_index] = src[src_start_index + i];
+            }
+        }
+    }
+
+    dst
+}
+
+mod test {
+    use crate::utils::bytes::{bytes_to_fields, fields_to_bytes};
+
+    #[test]
+    fn test_bytes_to_1_field() {
+        let input = [
+            1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31
+        ];
+        let output = bytes_to_fields::<31, 1>(input);
+
+        assert_eq(output[0], 0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f);
+    }
+
+    #[test]
+    fn test_1_field_to_bytes() {
+        let input = [0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f];
+        let output = fields_to_bytes::<31, 1>(input);
+
+        assert_eq(
+            output, [
+            1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31
+        ]
+        );
+    }
+
+    #[test]
+    fn test_bytes_to_2_fields() {
+        let input = [
+            1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59
+        ];
+        let output = bytes_to_fields::<59, 2>(input);
+
+        assert_eq(output[0], 0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f);
+        assert_eq(output[1], 0x202122232425262728292a2b2c2d2e2f303132333435363738393a3b);
+    }
+
+    #[test]
+    fn test_2_fields_to_bytes() {
+        let input = [
+            0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f, 0x202122232425262728292a2b2c2d2e2f303132333435363738393a3b
+        ];
+        let output = fields_to_bytes::<62, 2>(input);
+
+        assert_eq(
+            output, [
+            1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 0, 0, 0, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59
+        ]
+        );
+    }
+
+    #[test]
+    fn test_large_random_input_to_fields_and_back() {
+        let input = [
```

Comment:
How is a sequence from 1 to 128 random? :joy: I think Noir now has some light fuzzing mechanisms, perhaps we can use that.

Reply:
It'd also be good to test that starting from an arbitrary field array we can go into bytes and back. I know that's not what we're doing right now, but people might use it that way.

Reply:
fuzzing it was easier than expected https://github.com/AztecProtocol/aztec-packages/pull/8590/commits/1399cadcbcaa0721492d17156e9d183ac7331597

Reply:
> It'd also be good to test that starting from an arbitrary field array we can go into bytes and back. 

Addressed in https://github.com/AztecProtocol/aztec-packages/pull/8590/commits/0398c15d032bcd4726879952defb66f43d18e5be